### PR TITLE
fix: NaN when string is inputed on quantity input

### PR DESCRIPTION
**Pull request by EmanuelRamos115**
## What's the purpose of this pull request?

Resolves the bug that when typing letters in the quantity input a "NaN" error appears

## How it works?

I validated the function to change the input to remove any non-numeric value

## How to test it?

Go to the product page and try to enter a value other than a number in the quantity input

https://newswarovskiargentina-cm2bukmqh008u9fpvz46jty9h-rarbnwb8d.b.vtex.app/5662923-conjunto-stilla-tallas-mixtas-multicolor-bano-tono-oro-rosa-11436/p

## Checklist

**PR Title and Commit Messages**

- [x] PR title and commit messages follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification
  - Available prefixes: `feat`, `fix`, `chore`, `docs`, `style`, `refactor` and `test`

**PR Description**

- [x] Added a label according to the PR goal - `breaking change`, `bug`, `contributing`, `performance`, `documentation`..

**Dependencies**

- [x] Committed the `yarn.lock` file when there were changes to the packages

### CODE REVIEW

File: packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx [hunk 1]
```diff
@@ -139,6 +140,10 @@ const QuantitySelector = ({
         value={useUnitMultiplier ? multipliedUnit : quantity}
         onChange={changeInputValue}
         onBlur={validateBlur}
+        onInput={(event: React.FormEvent<HTMLInputElement>) => {
+          const input = event.currentTarget
+          input.value = input.value.replace(/\D/g, '')
+        }}
```
Comment (hellofanny) [inline packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx:146]: can't we use just `type="number"` here 🤔

Comment (EmanuelRamos115) [reply packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx:146]: Yeah, it could be... we would probably have the same result

Comment (hellofanny) [reply packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx:146]: actually, I think it’s safer to use with this validation ✅

**Review by hellofanny — APPROVED** [review]

File: packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx (hunk 1 above)
Comment (EmanuelRamos115) [reply packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx]: It seems to work well, but by adding this prop, the input appears with these little arrows and there is also a behavior that occurs when the user copies text and pastes it into the input, the value goes to 0, unlike the other approach, by js, which maintains the value that was entered, however it blocks any non-numeric character.

image

Comment (pedromtec) [reply packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx]: I don't know if It's a good idea to use type number if it changes de layout of the input.

Comment (hellofanny) [reply packages/components/src/molecules/QuantitySelector/QuantitySelector.tsx]: @Emanuel-Ramos sorry, I think the previous changes has a better approach 👍  Let's undo the latest changes (using `type=number`)

**Review by hellofanny — APPROVED** [review]
@EmanuelRamos115 thank you for the fix 🚀

**Pull request merged**